### ISSUE TITLE
Way to not include omitempty by default?

### ISSUE THREAD

**Issue by JoakimSoderberg** [go-swagger/go-swagger#2386]
## Problem statement

In https://github.com/go-swagger/go-swagger/pull/1189 support for adding `x-omitempty: false` to properties was added, so that the generated code for that particular property excludes `omitempty` in the `struct` defintion of the Model.

However, it's quite tedious having to add this to each and every property.

**Is there some way to disable `omitempty` globally when generating the code?**
Otherwise it would be nice it could be added.

**Comment by fredbi** [comment]
You are right. The way I see it, every `x-...` extension should be associated with a corresponding CLI option to make it systematic.  It should rather straightforward to implement it this way (the local extension would override the global setting).
If you feel like pushing a PR along these lines, I'd be happy to help out.

**Comment by JoakimSoderberg** [comment]
Hey, ok I tried looking at this in the code.

But I'm not sure where the best place would be to pass on a command line option like this is, so that I can pass it on to the template properly?

`IsEmptyOmitted` here:
https://github.com/go-swagger/go-swagger/blob/9a3827dd3741b3e90c2811c0a20cd79d383581fe/generator/types.go#L802-L814

Is what's used directly in the templates.

Any pointers?

**Comment by fredbi** [comment]
ok. So basically there are 3 things to change:
- the CLI options (here: https://github.com/go-swagger/go-swagger/blob/48f1315d12b5b1be444c4b5c197d61e00e7b981a/cmd/swagger/commands/generate/model.go#L31)
-the generator options set by the CLI: https://github.com/go-swagger/go-swagger/blob/48f1315d12b5b1be444c4b5c197d61e00e7b981a/generator/shared.go#L236
- the model & type resolution:
  * here (and probably a few other places in the same source file): https://github.com/go-swagger/go-swagger/blob/48f1315d12b5b1be444c4b5c197d61e00e7b981a/generator/types.go#L893
 * also here for some edge cases: https://github.com/go-swagger/go-swagger/blob/48f1315d12b5b1be444c4b5c197d61e00e7b981a/generator/model.go#L1645

**Comment by fredbi** [comment]
We probably need to convey some options to the type resolver.

**Comment by JoakimSoderberg** [comment]
Yes I had found all those spots you mentioned. I tried adding an option.

> We probably need to convey some options to the type resolver.

So this was the biggest crux for me. How do I access the option here:
https://github.com/go-swagger/go-swagger/blob/48f1315d12b5b1be444c4b5c197d61e00e7b981a/generator/types.go#L893-L901

The `resolvedType` type is created in multiple places as it looked to me, so no clear "constructor" where I could pass it to all instances.

**Comment by JoakimSoderberg** [comment]
@fredbi any hints on where to pass on the option?

**Comment by MJjainam** [comment]
@JoakimSoderberg @fredbi I was looking at this and wanted to contribute. 

Instead of passing it to all the instances as @JoakimSoderberg mentioned above, can we pass the option as part of the schema?

**Comment by MJjainam** [comment]
@fredbi 
I spent some time understanding and these are my findings on how to implement this

1. **We will need an additional CLI flag to pass global x-* options.**
    The reason we cannot use existing struct-tags is - whatever we pass in struct tags gets "added" in the model struct. If we want to remove something, that is not possible with struct-tags. 
    
2.  Coming to the difficult part, we somehow need to set `IsEmptyOmitted` attribute of Properties[i] under models. This can be achieved in two ways
     a. **Preprocess Schema**: `makeGenDefinitionHierarchy`  function is where the `Schema` is translated to `GenSchema`. So one way to pass in the x-* values is by modifying the schema.VendorExtensible before the translation. 
     b. **Postprocess GenSchema**: After the translation of `Schema` to `GenSchema` in function `makeGenDefinitionHierarchy`, we can edit the `GenSchema` object and set the x-* properties. 
     
@fredbi  @casualjim  @youyuanwu Could you guide what would be the best approach among the above two to implement this enhancement?

**Comment by lyuzaw** [comment]
Any updates on this issue?

**Comment by usmanovbf** [comment]
Hi there! is it still actual?
